pages/learn/more/anti-patterns.md

```diff
@@ -0,0 +1,96 @@
+---
+title: Development Anti-patterns
+---
+# How to break {{ $names.os.lower }} (development anti-patterns)
+
+As you are building your application and preparing for production, there are a number of things to keep in mind that can cause problems on a device. Many of these items may seem obvious or even desired, but often cause more trouble than they are worth.
+
+## Networking
+One of the biggest causes of issues during deployment is either a misconfigured local network or otherwise a network that does not adhere to the {{ $names.os.lower }} [networking requirements][networking-reqs].
+
+### Connectivity dependent on container (e.g. kernel module for wifi dongle, or udev rule for modem)
+Since there can be cases where application containers fail to start, having any networking configuration depend on those application containers without a fallback is generally discouraged. Using a project like [wifi-connect][wifi-connect] can help configure the networking stack after launch, but devices should be connected as reliably as possible at boot.
+
+### Iptables rules that block supervisor/VPN traffic
+It is enticing to use custom iptables rules to filter traffic before it hits the application, to limit the surface of the API. Those rules combined with host networking however can prevent the supervisor or VPN from successfully connecting to {{ $names.cloud.lower }}, and therefore should be combined with extreme scrutiny. For a full list of networking requirements, consult [this document][networking-reqs].
+
+### Running docker-in-docker with host networking and iptables enabled
+(can be folded into the previous issue, leaving for completeness at review)
+
+### Manually changing the system clock, or blocking NTP requests
+Since {{ $names.os.lower }} & the supervisor communicate with {{ $names.cloud.lower }} using an HTTP API, it is important that time is synchronized on the device to ensure that certificates are trusted. {{ $names.os.upper }} provides a [number of mechanisms][time-sync] to keep time as up-to-date as possible, but ensuring [NTP is accessible over the network][networking-reqs] is critical.
```

Comment:
* HTTP API → HTTPS API  (or "HTTP API over HTTPS", but sounds a bit pedantic...).  My point being that HTTPS (or SSL) are important to mention.
* certificates → SSL certificates
* certificates are trusted → certificate validation succeeds
* We know what it means (I think I do), :-) but users would benefit from expanding a bit on "why" time is relevant for SSL certificates. What about:

> Since {{ $names.os.lower }} and the supervisor communicate with {{ $names.cloud.lower }} using an HTTPS API, it is important that time is synchronized on the device. If the system date/time drifts substantially, SSL certificate validation may fail and the device may unexpectedly lose the ability to reach HTTPS websites or update the {{ $names.cloud.lower }} web dashboard, and may even no longer be reachable over ssh or the {{ $names.cloud.lower }} VPN. {{ $names.os.upper }} provides a [number of mechanisms][time-sync] to keep time as up-to-date as possible, but ensuring [NTP is accessible over the network][networking-reqs] is critical.

Reply:
Thanks for pointing these out, I think these are good changes. Updated!